osgp/shared/shared/src/main/java/org/opensmartgridplatform/shared/security/RsaEncrypter.java

```diff
@@ -0,0 +1,114 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.shared.security;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.security.Key;
+import java.security.KeyFactory;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.X509EncodedKeySpec;
+import java.util.Arrays;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+
+import org.opensmartgridplatform.shared.exceptionhandling.EncrypterException;
+
+//TODO merge with RsaEncryptionService; it is almost the same except for different configuration
+//  refactor it to 1 single configurable class with 2 instances with different configurations
+public class RsaEncrypter {
+    private static final int BLOCK_SIZE = 16;
+    private static final String ALG = "RSA";
+    private static final String ALGORITHM = "RSA/ECB/PKCS1Padding";
+
+    private Key publicKey;
+    private Key privateKey;
+
+    public void setPrivateKeyStore(File privateKeyStoreFile) throws EncrypterException {
+        try {
+            byte[] keyData = Files.readAllBytes(privateKeyStoreFile.toPath());
+            PKCS8EncodedKeySpec privateKeySpec = new PKCS8EncodedKeySpec(keyData);
+            this.privateKey = KeyFactory.getInstance(ALG).generatePrivate(privateKeySpec);
+        } catch (NoSuchAlgorithmException | IOException | InvalidKeySpecException e) {
+            throw new EncrypterException("Could not get cipher", e);
+        }
+    }
+
+    public void setPublicKeyStore(File publicKeyStoreFile) {
+        try {
+            byte[] keyData = Files.readAllBytes(publicKeyStoreFile.toPath());
+            X509EncodedKeySpec publicKeySpec = new X509EncodedKeySpec(keyData);
+            this.publicKey = KeyFactory.getInstance(ALG).generatePublic(publicKeySpec);
+        } catch (NoSuchAlgorithmException | IOException | InvalidKeySpecException e) {
+            throw new EncrypterException("Could not set public keystore", e);
+        }
+    }
+
+    protected Cipher getCipher() {
+        try {
+            return Cipher.getInstance(ALGORITHM);
+        } catch (NoSuchPaddingException | NoSuchAlgorithmException e) {
+            throw new EncrypterException("Could not get cipher", e);
+        }
+    }
+
+    protected Key getSecretEncryptionKey(int cipherMode) {
+        return cipherMode == Cipher.ENCRYPT_MODE ? this.publicKey : this.privateKey;
+    }
+
+    public byte[] encrypt(final byte[] secret) {
+        if(secret==null) {
+            throw new IllegalArgumentException("Can not encrypt NULL value");
+        }
+        try {
+            final Cipher cipher = this.getCipher();
+            cipher.init(Cipher.ENCRYPT_MODE, this.getSecretEncryptionKey(Cipher.ENCRYPT_MODE));
+            byte[] rsa = cipher.doFinal(secret);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "rsa". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1488)